### DISCUSSION TITLE
Why clippy always not happy? is the problem of the language or clippy?

### DISCUSSION THREAD

**Discussion by ghost** [rust-lang/rust-clippy#8351] [General]
If you search on github, you may found lots of commit message said: make clippy happy.
https://github.com/search?q=make+clippy+happy&type=commits
Why clippy always not happy?
Is it the problem of Rust lang, or the problem of the Clippy.

If each project can not make Rust clippy happy by default.
How dare Rust can ?
Why you just happy by default.

So may we say  Rust is a language like Trush?

If every one can not do the right thing with Rust, is it the fault of Rust lang ?

**Comment by llogiq** [comment]
Clippy is actually happy all the time. But clippy is especially happy to help. So seeing PRs made with its support (and yes, being a paperclip, clippy uses "it" pronoun) makes clippy exceptionally happy.

**Reply by ghost** [reply]
Now rust lang has the world largest lint rules in the world. and still growing fast. 

So many rules, you write 100 lines code, you got 500 bugs or warning .

Is it right?

No, no one can fight ,no one can win.

You just leave a week, you found we growing 50 new rules. 

You need catch up with it . and then another week, you got another 100 new lint rules.

**Reply by flip1995** [reply]
I just checked the history of our lint count in the `README` and it seems that Clippy adds 50 lints in a span of 6-10 months. (Not all lints are enabled by default though)

So what may feel like a week is more like a year.

**Reply by llogiq** [reply]
If you manage to get 500 warnings in 100 lines of code, you are a truly exceptional programmer!

Different programs may need different lints. Rust's user base is really diverse, so our lints follow.
